src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/ContactDiaryStoreV1.swift

```diff
@@ -50,10 +50,48 @@ class ContactDiaryStoreV1: DiaryStoring, DiaryProviding {
 		guard case .success = updateDiaryResult else {
 			return nil
 		}
+		
+		migrateColumns(inTable: "ContactPerson")
+		migrateColumns(inTable: "Location")
 
 		registerToDidBecomeActiveNotification()
 	}
-
+	
+	func migrateColumns(inTable tableName: String) {
+		databaseQueue.inDatabase { database in
+			let queryResult = database.prepare("PRAGMA table_info(" + tableName + ")" )
+			var finalSQL: String?
+			
+			while queryResult.next() {
+				let name = queryResult.string(forColumn: "name")
+				let type = queryResult.string(forColumn: "type")
+
+				// do migration for contact diary tables if the type of the Column "name" is "STRING"
+				if name == "name" && type == "STRING" {
+					finalSQL = """
+				    ALTER TABLE \(tableName) RENAME TO tmp;
+				    CREATE TABLE \(tableName) (
+				    id INTEGER PRIMARY KEY,
+				    name TEXT NOT NULL CHECK (LENGTH(name) <= 250)
+				    );
+				    INSERT INTO \(tableName)(id, name)
+				    SELECT id, name
+				    FROM tmp;
```

Comment:
Needed?

Reply:
To copy the data from the tmp table to the new one

Reply:
🤦‍♂️  I was expecting a semicolon in line 76, which does not exist. Indenting these two lines might help to understand the statement.